YubiKit/YubiKit/FIDO/FIDOInterface.swift

```diff
@@ -17,213 +17,434 @@ import OSLog
+    private func sendRequest(cmd: UInt8, payload: Data?) async throws(Error) {
+        let payloadData = payload ?? Data()
+
+        // INIT command uses broadcast channel, all others use assigned channel
+        let cid = (cmd == Self.hidCommand(.`init`)) ? CTAP.CID_BROADCAST : channelId
+
+        // Send init frame with first chunk of data
+        let initFrame = buildInitFrame(channelId: cid, command: cmd, payload: payloadData)
+        do {
+            try await connection.send(initFrame)
+        } catch {
+            throw .connectionError(error)
+        }
+        /* Fix trace: trace(message: "Sent init frame (\(payloadData.count) total bytes)") */
+
+        // If payload is larger than init frame can hold, send continuation frames
+        var remainingData = payloadData.dropFirst(CTAP.INIT_DATA_SIZE)
+        var sequence: UInt8 = 0
+
+        while !remainingData.isEmpty {
+            let contFrame = buildContinuationFrame(
+                channelId: cid,
+                sequence: sequence,
+                payload: Data(remainingData)
+            )
+            do {
+                try await connection.send(contFrame)
+            } catch {
+                throw .connectionError(error)
+            }
+            /* Fix trace: trace(message: "Sent continuation frame \(sequence)") */
+
+            remainingData = remainingData.dropFirst(CTAP.CONT_DATA_SIZE)
+            sequence += 1
 
-        if (capabilities & Capability.WINK) != 0 {
-            /* Fix trace: trace(message: "Device supports WINK") */
+            guard sequence < 128 else {
+                throw Error.illegalArgument("Payload too large: exceeds maximum CTAP HID message size")
+            }
         }
     }
 
-    // MARK: - Private Helpers
+    /// Receive CTAP response (init frame + continuation frames if needed)
+    /// Handles KEEPALIVE messages
+    private func receiveResponse(expectedCommand: UInt8) async throws(Error) -> Data {
+        // Loop to handle KEEPALIVE messages
+        while true {
+            // Read init frame with frame timeout
+            let responseInitFrame: Data?
+            do {
+                responseInitFrame = try await withTimeout(frameTimeout) { try await self.connection.receive() }
+            } catch {
+                throw .connectionError(error as! FIDOConnectionError)
+            }
+
+            guard let responseInitFrame = responseInitFrame else {
+                throw .timeout()
+            }
+
+            let (responseChannelId, responseCommand, payloadLength, initFrameData) = try parseInitFrame(
+                responseInitFrame
+            )
+
+            // Validate channel ID (INIT responses can come from any channel)
+            if expectedCommand != Self.hidCommand(.`init`) {
+                guard responseChannelId == channelId else {
+                    throw Error.responseParseError("Invalid channel ID in response")
+                }
+            }
```

Comment:
The INIT response should come on the broadcast channel, so the check should be applicable there too (with channelId being initialized to the broadcast channel). No special handling should be needed here either.